core/include/detray/utils/grid/detail/axis.hpp

```diff
@@ -401,6 +408,25 @@ class multi_axis {
                                detray::get_data(m_edges)};
     }
 
+    /// Equality operator
+    ///
+    /// @param rhs the right-hand side of the comparison
+    ///
+    /// @note in the non-owning case, we compare the values not the pointers
+    ///
+    /// @returns whether the two axes are equal
+    DETRAY_HOST_DEVICE constexpr auto operator==(const multi_axis &rhs) const
+        -> bool {
+        if constexpr (!std::is_pointer_v<edge_range_t>) {
+            return m_edge_offsets == rhs.m_edge_offsets &&
+                   m_edges == rhs.m_edges;
+        } else {
+            return m_edge_offsets == rhs.m_edge_offsets &&
+                   *m_edges == *rhs.m_edges;
```

Comment:
I think, this compares the entire edges container, so also edges that belong to other multi axes. Are the equals operators of the single axis used anywhere outside their unittests?

Reply:
I see, that's not really what we wanna do ...

Reply:
Just one question: in the owning case, does every `multi_axis` own then a copy of the full edge container?

Reply:
No, because the owning case is for standalone grids (e.g. in the case of seeding or grids construction), while non-owning grids are used in the detector. Whenever we have multiple grids in global containers, the data is owned by the `grid_collection` instead

Reply:
Isn't the code ok then? Because the edges check is only for the non-owning case ...